pkg/apis/componentconfig/edgecore/v1alpha2/types.go

```diff
@@ -920,6 +920,9 @@ type EventBus struct {
 	// skip checking other EventBus configs.
 	// default true
 	Enable bool `json:"enable"`
+	// Enable indicates whether start mqtt container when edgecore start
+	// default true
+	EnableMqttContainer bool `json:"enableMqttContainer"`
```

Comment:
As a workaround solution, it's not needed to add new api field, can we assume that if evetbus module is enabled and mode is `external`, then we  need to save the `mqtt container`, and start it.

Reply:
> As a workaround solution, it's not needed to add new api field, can we assume that if evetbus module is enabled and mode is `external`, then we need to save the `mqtt container`, and start it.

I hope the --with-mqtt config in keadm can take effect, maybe I can set mode=internal when --with-mqtt=false?

Reply:
There're two scenarios users deploy edgecore:

1. Users don't use keadm to deploy edgecore, when they enable `eventbus`, they will deploy the mqtt by self
2. Users use keadm to deploy edgecore, when they set --with-mqtt=true, we need to deploy mqtt, or no need to deploy

If we can set a environment value in edgecore, like `WITH-MQTT`, default set as false, edgecore will not deploy mqtt, if uses use keadm to deploy, and set --with-mqtt=true, set the `WITH-MQTT` to true, and deploy mqtt with edgecore.

Reply:
> There're two scenarios users deploy edgecore:
> 
> 1. Users don't use keadm to deploy edgecore, when they enable `eventbus`, they will deploy the mqtt by self
> 2. Users use keadm to deploy edgecore, when they set --with-mqtt=true, we need to deploy mqtt, or no need to deploy
> 
> If we can set a environment value in edgecore, like `WITH-MQTT`, default set as false, edgecore will not deploy mqtt, if uses use keadm to deploy, and set --with-mqtt=true, set the `WITH-MQTT` to true, and deploy mqtt with edgecore.

Fixed. PTAL  @fisherxu  

![image](https://github.com/kubeedge/kubeedge/assets/48788150/67f1c51d-2c42-4421-b3e1-d24521e93a97)